#VRML_SIM R2025a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/robots/kuka/youbot/protos/Youbot.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/factory/containers/protos/PlasticFruitBox.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/factory/containers/protos/WoodenBox.proto"

WorldInfo {
  info [
    "GRASP TEST V3 - Multi-Angle Validation"
    "Tests grasp at 0°, ±30°, ±60° angles"
  ]
  title "YouBot Grasp Test V3"
  basicTimeStep 16
  contactProperties [
    ContactProperties {
      material1 "InteriorWheelMat"
      coulombFriction [
        0, 2, 0
      ]
      frictionRotation -0.785398 0
      bounce 0
      forceDependentSlip [
        10, 0
      ]
    }
    ContactProperties {
      material1 "ExteriorWheelMat"
      coulombFriction [
        0, 2, 0
      ]
      frictionRotation 0.785398 0
      bounce 0
      forceDependentSlip [
        10, 0
      ]
    }
  ]
}
Viewpoint {
  orientation -0.5773502691896258 0.5773502691896258 0.5773502691896258 2.0944
  position -0.5112727273139026 -0.060619736078962844 13.210532001517056
  follow "youBot"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation -0.79 0 0
  floorSize 7 4
  floorAppearance PBRAppearance {
    baseColorMap ImageTexture {
      url [
        "../textures/Material.001_baseColor.png"
      ]
    }
    roughness 1
    metalness 0
  }
  wallHeight 0.3
  wallAppearance PBRAppearance {
    baseColorMap ImageTexture {
      url [
        "../textures/Material.001_baseColor.png"
      ]
    }
    roughness 1
    metalness 0
  }
}
PlasticFruitBox {
  name "GREEN"
  translation 0.48 1.58 0
  color 0 1 0
}
PlasticFruitBox {
  name "BLUE"
  translation 0.48 -1.62 0
  color 0 0 1
}
PlasticFruitBox {
  name "RED"
  translation 2.31 0.00999969 0
  rotation 0 0 1 1.5708
  color 1 0 0
}
DEF YOUBOT Youbot {
  translation -1.5 0.0 0.102
  rotation 0 0 1 0
  controller "youbot_grasp_test_v3"
  controllerArgs [
    ""
  ]
  bodySlot [
    Lidar {
      translation 0.28 0 -0.07
      numberOfLayers 1
    }
    Camera {
      translation 0.27 0 -0.06
      width 128
      height 128
    }
  ]
}
Robot {
  name "SupervisorV3"
  controller "supervisor_test_v3"
  supervisor TRUE
}
WoodenBox {
  translation 0.6 0 0.15
  name "A"
  size 0.3 0.3 0.3
}
WoodenBox {
  translation 1.96 -1.24 0.15
  name "B"
  size 0.3 0.3 0.3
}
WoodenBox {
  translation 1.95 1.25 0.15
  name "C"
  size 0.3 0.3 0.3
}
WoodenBox {
  translation -2.28 1.5 0.15
  name "D"
  size 0.3 0.3 0.3
}
WoodenBox {
  translation -1.02 0.75 0.15
  name "E"
  size 0.3 0.3 0.3
}
WoodenBox {
  translation -1.02 -0.74 0.15
  name "F"
  size 0.3 0.3 0.3
}
WoodenBox {
  translation -2.27 -1.51 0.15
  name "G"
  size 0.3 0.3 0.3
}
